contracts/HumaniqToken.sol

```diff
@@ -63,76 +70,80 @@ contract HumaniqToken is StandardToken, SafeMath {
-        returns (bool success)
-    {
-        return super.transfer(_to, _value);
-    }
-
-    function transferFrom(address _from, address _to, uint256 _value)
-        unlocked
-        returns (bool success)
-    {
-        return super.transferFrom(_from, _to, _value);
-    }
-
     /// @dev Function to change address that is allowed to do emission.
     /// @param newAddress Address of new emission contract.
-    function changeEmissionContractAddress(address newAddress)
-        external
+    function changeMinter(address newAddress)
+        public
         onlyFounder
         returns (bool)
     {
-        emissionContractAddress = newAddress;
+        minter = newAddress;
+
+        // Allow emission contract to distribute tokens minted during ICO stage
+        allowed[allocationAddressICO][minter] = ICOSupply;
     }
 
-    /// @dev Function that locks/unlocks transfers of token.
-    /// @param value True/False
-    function lock(bool value)
-        external
+    /// @dev Function to change founder address.
+    /// @param newAddress Address of new founder.
+    function changeFounder(address newAddress)
+        public
         onlyFounder
+        returns (bool)
     {
-        locked = value;
+        founder = newAddress;
+
+        // Allow founder to distribute tokens minted during preICO stage
+        allowed[allocationAddressPreICO][founder] = preICOSupply;
     }
 
-    /// @dev Finalization of maxTotalSupply after the ICO.
-    function finalizeMaxTotalSupply(uint coinsIssued)
-      external
-      isCrowdfundingContract
-      returns (bool)
+    /// @dev Function to change multisig address.
+    /// @param newAddress Address of new multisig.
+    function changeMultisig(address newAddress)
+        public
+        onlyFounder
+        returns (bool)
     {
-      if (icoFinished) {
-        return false;
-      }
-      icoFinished = true;
-      // Set max total supply of coins to be 5X after an ICO.
-      maxTotalSupply = mul(coinsIssued, 5);
-      return true;
+        multisig = newAddress;
     }
 
     /// @dev Contract constructor function sets initial token balances.
-    /// @param _founder Address of the founder of HumaniQ.
-    function HumaniqToken(address _founder)
-    {
-        totalSupply = 0;
-        founder = _founder;
+    function HumaniqToken(address founderAddress)
+    {   
+        // Set founder address
+        founder = founderAddress;
+
+        // Allocate all created tokens during ICO stage to allocationAddressICO.
+        balances[allocationAddressICO] = ICOSupply;
+
+        // Allocate all created tokens during preICO stage to allocationAddressPreICO.
+        balances[allocationAddressPreICO] = preICOSupply;
+
+        // Allow founder to distribute tokens minted during preICO stage
+        allowed[allocationAddressPreICO][founder] = preICOSupply;
+
+        // Give 14 percent of all tokens to founders.
+        balances[multisig] = div(mul(ICOSupply, 14), 86);
```

Comment:
As you mentioned it, preICOSupply already contains this bonus, and therefore bonus tokens from preICO will be transferred to multisig address during migration of all balances from preICO token.